File: services/identity-service/src/main/java/br/com/deltaglobalbank/identity/features/tenants/createTenant/CreateTenantRequest.java
package br.com.deltaglobalbank.identity.features.tenants.createTenant;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

public record CreateTenantRequest(
    @NotBlank(message = "name é obrigatório")
    @Size(min = 3, max = 255, message = "name deve ter entre 3 e 255 caracteres")
    String name,

    @NotBlank(message = "slug é obrigatório")
    String slug
) {
}

File: services/identity-service/src/main/kotlin/br/com/deltaglobalbank/identity/features/tenants/createTenant/CreateTenantRequest.kt
package br.com.deltaglobalbank.identity.features.tenants.createTenant

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

data class CreateTenantRequest(
    @field:NotBlank(message = "name é obrigatório")
    @field:Size(min = 3, max = 255, message = "name deve ter entre 3 e 255 caracteres")
    val name: String,

    @field:NotBlank(message = "slug é obrigatório")
    val slug: String
)